icu4c/source/test/cintltst/cdateintervalformattest.c

```diff
@@ -380,4 +382,75 @@ static void TestFormatToResult() {
     udtitvfmt_closeResult(fdi);
 }
 
+static void TestFormatCalendarToResult() {
```

Comment:
Can you write a test where the calendar in the locale is different from the calendar in the UCalendar objects, which is the main point of adding this new API?